src/components/frame/v5/pages/UserProfilePage/UserProfilePage.tsx

```diff
@@ -1,59 +1,88 @@
-import React, { FC } from 'react';
-import { defineMessages } from 'react-intl';
-import { Navigate } from 'react-router-dom';
-
-import LoadingTemplate from '~frame/LoadingTemplate';
-import { useAppContext } from '~hooks';
-import { LANDING_PAGE_ROUTE } from '~routes';
-import { Form } from '~shared/Fields';
-
-import { useUserProfile } from './hooks';
-import UserProfilePageForm from './partials/UserProfilePageForm';
-import { FormValues, validationSchema } from './validation';
+import React, { FC, useEffect, useState } from 'react';
+import {
+  Outlet,
+  useLocation,
+  useNavigate,
+  useResolvedPath,
+} from 'react-router-dom';
+import { formatText } from '~utils/intl';
+import { usePageHeadingContext, useSetPageHeadingTitle } from '~context';
+import {
+  USER_ADVANCED_ROUTE,
+  USER_EDIT_PROFILE_ROUTE,
+  USER_PREFERENCES_ROUTE,
+} from '~routes';
+import Tabs from '~shared/Extensions/Tabs';
+import { tabRoutes } from './consts';
+import { TabId } from './types';
 
 const displayName = 'v5.pages.UserProfilePage';
 
-const MSG = defineMessages({
-  loadingText: {
-    id: `${displayName}.loadingText`,
-    defaultMessage: 'Loading user profile...',
-  },
-});
-
 const UserProfilePage: FC = () => {
-  const { user, userLoading, walletConnecting } = useAppContext();
+  const navigate = useNavigate();
+  const { pathname } = useLocation();
+  const resolvedEditProfilePath = useResolvedPath(USER_EDIT_PROFILE_ROUTE);
+  const resolvedPreferencesPath = useResolvedPath(USER_PREFERENCES_ROUTE);
+  const resolvedAdvancedPath = useResolvedPath(USER_ADVANCED_ROUTE);
+  const [activeTab, setActiveTab] = useState<TabId>(TabId.Tab0);
```

Comment:
Could you please use descriptive names for the tabs (enum members)? They seem to be Profile, Preferences, Advanced.